tests/solana_tests/primitives.rs

```diff
@@ -941,10 +948,14 @@ fn test_overflow_detect_unsigned() {
             let limit = BigUint::from(2_u32).pow(width);
 
             // Generate a random number within the the range [0, 2^N -1]
-            let first_operand_rand = rng.gen_biguint(width.into()).add(1_u32);
+            let mut first_operand_rand = rng.gen_biguint(width.into());
+
+            if first_operand_rand == BigUint::from(0_u32) {
+                first_operand_rand = BigUint::from(1_u32);
+            }
```

Comment:
What number multiplied by 1 overflows?

Reply:
presumably you want:

```rust
let first_operand_rand = rng.gen_biguint_range(&BigUint::from(2usize), &limit);
```

Reply:
Limit is then divided by this. 
second_op_rand = (limit / 1) + 2 
second_op * first_op will indeed overflow